### ISSUE TITLE
Dataselect exists with code 0 when nothing written due to chmod issue

### ISSUE THREAD

**Issue by heavelock** [iris-edu/dataselect#5]
Hi, 
I'm just fiddling with dataselect and I found a bug.
Let's say I'm executing that command:
```
$ ~/tools/dataselect/dataselect -v -sum -Pr -SDS /mnt/D4To/temp/tempsds SI.SI30.00.CHE_20190701_??0000.miniseed
```
The output dir has chmod 555
``` 
$ ls /mnt/D4To/temp
dr-xr-xr-x  3 damian damian 4,0K juil.  4 18:39 tempsds
```

In result, execution of that command is successful, dataselect returns code 0. Unfortunately, none of the files is written.
Here is cutout of the log:

```
...
Reading: SI.SI30.00.CHE_20190701_230000.miniseed
Files: 24, Records: 75288, Samples: 21600000
Pruning trace data
Writing output data
cannot open data stream file, Permission denied
Wrote 1131008 bytes from file SI.SI30.00.CHE_20190701_000000.miniseed
...
Wrote 38547456 bytes of 75288 records to output file(s)                                                                                                                                                                                                                                                                                                                                       
```

**Comment by chad-iris** [comment]
Thanks @heavelock!  Fixed in version 3.22

**Comment by heavelock** [comment]
Thanks!
